### PR TITLE
Port WriteOutputFiles

### PR DESCRIPTION
**Pull request by caseypt**
Like #28, but rebased on #30 and without the various model fixes and temporary changes.

**Testing**
* Run the model with the sample GMS file. `python ./run.py sample_4.gms`.
* The model should complete successfully and you should get output that looks like [this](https://gist.github.com/caseypt/a8b021689d47bc9f914eadd6a35e49ad).

~~Issues with the `ManNitr` and `ManPhos` variables #25~~ Fixed by #30 (3f776d8017e95e774361967d73be735ed73dfa7c).
Port PrelimQualCalculations #26 
~~CalcCN index errors #27~~ Fixed by #30 (https://github.com/WikiWatershed/gwlf-e/pull/30/commits/2ea2c464c7f1eddb61c973475708e889e4cffc99)

Connects to #18.

### CODE REVIEW

**Comment by caseypt** [comment]
@kdeloach cd3410e addresses comments made in #30.

**Comment by caseypt** [comment]
After the changes in #30 were implemented, `AvRunoff` is now populated in the output: https://gist.github.com/caseypt/a3bdd8e73c8799ecbd5c5dc191ce2881

Old output: https://gist.github.com/caseypt/a8b021689d47bc9f914eadd6a35e49ad

File: gwlfe/LoadReductions.py [hunk 1]
```diff
@@ -3,6 +3,340 @@
...
+    NROWBMP6 = (z.n28b / 100) * z.n5 * z.n70
+    NROWNM = z.n5 - NROWBMP6
+    NROWBMP1 = (z.n25 / 100) * NROWNM * z.n63
+    NROWBMP2 = (z.n26 / 100) * NROWNM * z.n65
+    NROWBMP3 = (z.n27 / 100) * NROWNM * z.n66
+    NROWBMP4 = (z.n27b / 100) * NROWNM * z.n66b
+    NROWBMP5 = (z.n28 / 100) * NROWNM * z.n67
+    NROWBMP8 = (z.n29 / 100) * NROWNM * z.n68
+    NROWRED = NROWNM - (NROWBMP1 + NROWBMP2 + NROWBMP3 + NROWBMP4 + NROWBMP5 + NROWBMP8)
+    if (z.SMCheck == "Both" or z.SMCheck == "Uplaz.nd"):
```
Comment (kdeloach) [inline gwlfe/LoadReductions.py]: Typo "Uplaz.nd"?

Comment (kdeloach) [inline gwlfe/LoadReductions.py]: Err... I guess this variable is hard-coded to "Both" in any case.

**Comment by kdeloach** [comment]
+1 I have output.

> {u'monthly': [{u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 9.3818435946261509e+102, u'AvRunoff': 9.3771295258859806e+102, u'AvGroundWater': 6.2178192352835493e+99, u'AvPtSrcFlow': 5.4371470967657236e+99, u'AvWithdrawal': 1.3391807565623798e+101, u'AvStreamFlow': 9.2548664165617909e+102}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 733.78599999998801, u'AvRunoff': 722.53028270912, u'AvGroundWater': 716.62646065345268, u'AvPtSrcFlow': 0.99929981991558692, u'AvWithdrawal': 27.250066666667017, u'AvStreamFlow': 1412.9059765158215}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 2145.5019999999786, u'AvRunoff': 2146.8951031929296, u'AvGroundWater': 1081.7806519619601, u'AvPtSrcFlow': 2.0619746515296344, u'AvWithdrawal': 50.786866666667244, u'AvStreamFlow': 3179.9508631397521}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 2323.6346666666409, u'AvRunoff': 2309.3450637394267, u'AvGroundWater': 269.76201805250076, u'AvPtSrcFlow': 2.0180856330509962, u'AvWithdrawal': 51.362733333336223, u'AvStreamFlow': 2529.7624340916427}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 1951.6113333333521, u'AvRunoff': 1949.0639999999919, u'AvGroundWater': 41.284444846518831, u'AvPtSrcFlow': 1.8912818248415673, u'AvWithdrawal': 46.582666666664814, u'AvStreamFlow': 1945.6570600046875}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 2113.4533333333179, u'AvRunoff': 2112.235333333269, u'AvGroundWater': 7.0037270296033221, u'AvPtSrcFlow': 1.5195737608379305, u'AvWithdrawal': 38.67500000000031, u'AvStreamFlow': 2082.08363412371}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 1671.6926666666316, u'AvRunoff': 1670.5739999999896, u'AvGroundWater': 1.9615554715404229, u'AvPtSrcFlow': 1.3538523621924867, u'AvWithdrawal': 33.345666666667306, u'AvStreamFlow': 1640.5437411670553}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 1153.6233333333569, u'AvRunoff': 1152.2420000000127, u'AvGroundWater': 1.3945229227472851, u'AvPtSrcFlow': 1.080732471338036, u'AvWithdrawal': 26.618666666667352, u'AvStreamFlow': 1128.0985887274305}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 1160.0746666666589, u'AvRunoff': 1159.2853333333421, u'AvGroundWater': 0.88955523647282941, u'AvPtSrcFlow': 0.76047447035602578, u'AvWithdrawal': 19.354999999999595, u'AvStreamFlow': 1141.5803630401715}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 748.17333333333443, u'AvRunoff': 747.52333333332979, u'AvGroundWater': 0.58815328655768462, u'AvPtSrcFlow': 0.54330300868895598, u'AvWithdrawal': 13.381666666666812, u'AvStreamFlow': 735.27312296190973}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 230.23266666666629, u'AvRunoff': 225.87960660950671, u'AvGroundWater': 1.9069757981707491, u'AvPtSrcFlow': 0.25000770110438419, u'AvWithdrawal': 6.3630000000000102, u'AvStreamFlow': 221.67359010878184}, {u'AvEvapoTrans': 0.0, u'AvTileDrain': 0.0, u'AvPrecipitation': 66.70999999999998, u'AvRunoff': 59.955678594782604, u'AvGroundWater': 6.7529842492321457, u'AvPtSrcFlow': 0.053525056653749889, u'AvWithdrawal': 1.3183333333333334, u'AvStreamFlow': 65.44385456733518}], u'meta': {u'NRur': 10, u'SedDelivRatio': 0.149, u'NLU': 16, u'NUrb': 6, u'WxYrBeg': 2000, u'NYrs': 15, u'WxYrEnd': 2014}}

File: gwlfe/LoadReductions.py (hunk 1 above)
Comment (caseypt) [inline gwlfe/LoadReductions.py]: Good catch. I'll fix it.

**Pull request merged**